Hometask4.playground/Contents.swift

```diff
@@ -0,0 +1,84 @@
+
+import Foundation
+
+var worldPlaceDictionary = ["Russia": ("The biggest country in the world", "Eurasia"),
+                            "Portugal": ("One of the most safety country in the world","Europe"),
+                            "Australia": ("The same as Europe but without migrants from near East","Australia"),
+                            "Netherlands": ("Gaaga", "Europe"),
+                            "Belarus": ("Misnk is one of the most clean city", "Europe"),
+                            "Taiwan": ("The main center of microconductor manufactor", "Asia"),
+                            "Japan": ("Neutral country", "Asia"),
+                            "USA": ("Top 10 companies with the biggest capitalisation placed in USA", "North America")]
+
+//1. Добавить новый элемент в словарь, поменять два существующих элемента в словаре, удалить один элемент из словаря.
+
+worldPlaceDictionary["Minsk"] = ("1067 battle on Nemiga", "Europe") // add a new element
+
+worldPlaceDictionary["Russia"]    = ("Putinka", "Eurasia")
+worldPlaceDictionary["Australia"] = ("A1", "Australia") // change elements in dictionary
+
+worldPlaceDictionary.removeValue(forKey: "Minsk")
+worldPlaceDictionary["Netherlands"] = nil // remove value from the Dictionary
+
+//2. Создать отдельный словарь с названием городов/стран с длиной более 6 букв из первоначального словаря.
+
+var dictCitiesLongName = [String: (String, String)] ()
+
+dictCitiesLongName = worldPlaceDictionary.filter({ $0.key.count > 6 })
+
+//3.  Создать массив интересных фактов в формате (“название города/страны: факт”) из первоначального словаря, отсортировав их по длине строки по убыванию
+
+var countriesFactsArray = [String]()
+
+for (country, fact) in worldPlaceDictionary {
+    countriesFactsArray.append("\(country): \(fact.0)")
+    countriesFactsArray = countriesFactsArray.sorted(by: { $0.count > $1.count })
+}
+
+countriesFactsArray = worldPlaceDictionary.map { "\($0.key): \($0.value.0) "}.sorted(by: { $0.count > $1.count })
+
+//4.* Отсортировать страны/города по континенту. Отсортированные данные должны лежать в одном объекте (сами придумайте, в каком). В итоге, у вас должна быть возможность получить все страны/города переданному континенту. (Лучше тут сразу функцию реализовать, которая будет возвращать страны по переданному континенту. Чтобы я потом не задавалась, вопросом, как мне потом страны получить из вашего объекта)
+
+var sortedArrayByCountries = [String : [String]] ()
+
+for key in worldPlaceDictionary {
+    sortedArrayByCountries[key.value.1, default: []] += [key.key]
+}
+print(sortedArrayByCountries["Europe"]  ?? 0) /* Honestly, I don't understand what this does at all [key.value.1, default: [] += [key.key]
+I found the answer to my question in the lecture, but it's still unclear to me. It seems to me that something with my head , simple things are difficult */
```

Comment:
`sortedArrayByCountries[key.value.1]` returns an optional array of `String`([String]?). We want to add a new element to this array. We need to unwrap this optional. If the value is `nil` we need to create an array for that key and add `key.key` into this array. If the value isn't nil, we need to add `key.key` into this array found array. But we can do it much easier using `default`. You need to set here a value that would be returned if the value is nil. So `sortedArrayByCountries[key.value.1, default: []]` always returns `[String]` and you can add an element here straight away without unwrapping and all following stuff

Reply:
Hm , now it is clear for me , i need more practice with this thing to understand it at all

Reply:
I found the "Swift Taskbook (language version 4.0)".
Can I use it in combination with codewars? Can I ask you to check my with the tasks from "Taskbook"?

Reply:
Yeah, sure